src/dex/lite-psm/lite-psm.ts

```diff
@@ -490,51 +492,59 @@ export class LitePsm
     return { params: payload, encoder, networkFee: '0' };
   }
 
+  async updatePoolState(): Promise<void> {
+    const blockNumber = await this.dexHelper.web3Provider.eth.getBlockNumber();
+
+    await this.initializeEventPools(blockNumber, false);
+  }
+
   // Returns list of top pools based on liquidity. Max
   // limit number pools should be returned.
   async getTopPoolsForToken(
     tokenAddress: Address,
     limit: number,
   ): Promise<PoolLiquidity[]> {
-    const _tokenAddress = tokenAddress.toLowerCase();
+    const token = tokenAddress.toLowerCase();
+
+    const isDAI = this.dai.address.toLowerCase() === token;
+    const isUSDS = this.usds.address.toLowerCase() === token;
+    const isDAIorUSDS = isDAI || isUSDS;
+
     // Liquidity depends on the swapping side hence we simply use the min
     // Its always in terms of stable coin hence liquidityUSD = liquidity
     const minLiq = (poolState: PoolState, decimals: number) => {
       const buyLimit = poolState.gemBalance * getBigIntPow(18 - decimals);
       const sellLimit = poolState.daiBalance;
-      return (
-        2 *
-        parseInt(
-          (
-            (buyLimit > sellLimit ? sellLimit : buyLimit) / BI_POWS[18]
-          ).toString(),
-        )
-      );
-    };
 
-    const isDaiOrUsds =
-      _tokenAddress === this.dai.address || _tokenAddress === this.usds.address;
+      const minLimit = buyLimit < sellLimit ? buyLimit : sellLimit;
 
-    const validPoolConfigs = isDaiOrUsds
+      const liquidity = Number(minLimit) / Number(BI_POWS[18]);
+      return liquidity > 0 ? 2 * liquidity : 0;
+    };
+
+    const validPoolConfigs: PoolConfig[] = isDAIorUSDS
       ? this.poolConfigs
-      : this.eventPools[_tokenAddress]
-      ? [this.eventPools[_tokenAddress].poolConfig]
+      : this.eventPools[token]
+      ? [this.eventPools[token].poolConfig]
       : [];
+
     if (!validPoolConfigs.length) return [];
 
-    const poolStates = await getOnChainState(
-      this.dexHelper.multiContract,
-      validPoolConfigs,
-      this.vatAddress,
-      'latest',
-      LitePsmConfig[this.dexKey][this.network].dai.address,
-    );
+    const pools = validPoolConfigs.map(config => {
+      const eventPoolKey = config.gem.address.toLowerCase();
+      const eventPool = this.eventPools[eventPoolKey];
+      const state = eventPool?.getStaleState();
+
+      if (!state) return null;
+
+      return {
+        exchange: this.dexKey,
+        address: config.psmAddress,
+        liquidityUSD: minLiq(state, config.gem.decimals),
+        connectorTokens: isDAI ? [config.gem] : [this.dai, this.usds],
```

Comment:
In previous implementation it was `isDaiOrUsds`, in current implementation it's only `isDAI`.Assume we should stick to `isDaiOrUsds`